packages/ui/src/primitives/icons/icon.tsx

```diff
@@ -1,12 +1,13 @@
 import * as React from 'react';
+import { mergeProps } from 'react-aria';
 
 export interface IconProps extends React.SVGProps<SVGSVGElement> {}
 
 export const Icon = React.forwardRef<SVGSVGElement, IconProps & { children: React.ReactElement }>(
   function Icon(props, ref) {
     const { children, 'aria-label': ariaLabel, 'aria-hidden': ariaHidden, ...restProps } = props;
     return React.cloneElement(children, {
-      ...restProps,
+      ...mergeProps(children.props, restProps),
```

Comment:
This addition allows the icon to have a class and the implementation of the icon to have a class that is merged. This allows us to have `className='rtl:rotate-180'` on the arrow icon so it always respects rtl support.